File: QE-Java Selenium/Activity17.java
package demos;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.JavascriptExecutor;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

public class Activity17 {
    public static void main(String[] args) throws InterruptedException {

        WebDriver d = new FirefoxDriver();
        d.get("https://training-support.net/webelements/selects");
        System.out.println("Title: " + d.getTitle());

        Select m = new Select(d.findElement(
            By.xpath("/html/body/div/main/div/div/div/div/section/div[2]/select")
        ));

        m.selectByVisibleText("HTML");
        m.selectByIndex(3);
        m.selectByIndex(4);
        m.selectByIndex(5);

        JavascriptExecutor js = (JavascriptExecutor) d;

        js.executeScript("arguments[0].selected=true;",
            d.findElement(By.xpath(
            "/html/body/div/main/div/div/div/div/section/div[2]/select/option[11]"))
        );

        js.executeScript("arguments[0].selected=false;",
            d.findElement(By.xpath(
            "/html/body/div/main/div/div/div/div/section/div[2]/select/option[5]"))
        );

        Thread.sleep(1000);
        d.quit();
    }
}

File: QE-Python Selenium/Activity17.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import Select
import time

d = webdriver.Firefox()
d.get("https://training-support.net/webelements/selects")
print("Title:", d.title)

m = Select(d.find_element(By.XPATH,
    "/html/body/div/main/div/div/div/div/section/div[2]/select"))

m.select_by_visible_text("HTML")
m.select_by_index(3)
m.select_by_index(4)
m.select_by_index(5)

d.execute_script("arguments[0].selected=true;",
    d.find_element(By.XPATH,
    "/html/body/div/main/div/div/div/div/section/div[2]/select/option[11]"))

d.execute_script("arguments[0].selected=false;",
    d.find_element(By.XPATH,
    "/html/body/div/main/div/div/div/div/section/div[2]/select/option[5]"))

time.sleep(1)
d.quit()
